### ISSUE TITLE
Support for built-in collection types

### ISSUE THREAD

**Issue by a-shkarupin** [mitchelllisle/sparkdantic#184]
Hey,
using python3.10, sparkdantic==0.17.0, pydantic==2.5.3, pydantic_core==2.14.6

```python
from sparkdantic import SparkModel


class FooBar(SparkModel):
    id: int
    name: str
    words: list[str]



print(FooBar.model_spark_schema())
```

Getting following error:
```
Traceback (most recent call last):
  File "/home/alsh/tmp/code.py", line 11, in <module>
    print(FooBar.model_spark_schema())
  File "/home/alsh/Envs/sparkdantic-reproduce/lib/python3.10/site-packages/sparkdantic/model.py", line 238, in model_spark_schema
    if cls._is_spark_model_subclass(t):
  File "/home/alsh/Envs/sparkdantic-reproduce/lib/python3.10/site-packages/sparkdantic/model.py", line 273, in _is_spark_model_subclass
    return (inspect.isclass(value)) and (issubclass(value, SparkModel))
  File "/usr/lib/python3.10/abc.py", line 123, in __subclasscheck__
    return _abc_subclasscheck(cls, subclass)
TypeError: issubclass() arg 1 must be a class
```

The code works if I import `typing.List` and use it instead. Would it be possible to add support for built-in collection types?

**Comment by mitchelllisle** [comment]
HI, yep this is definitely a bug. Will work on this when I get some time this week

**Comment by mitchelllisle** [comment]
Fixed in #187
